### PR TITLE
Add make commands to generate configuration docs

### PR DESCRIPTION
**Pull request by lornajane**
Based heavily on https://github.com/aprepo/aiven-doc-generator (thanks @aprepo)
Generates include files that can be referred to from the main content,
and regenerated as needed.

### CODE REVIEW

File: docs/products/kafka/reference/advanced-params.rst [hunk 1]
```diff
@@ -0,0 +1,6 @@
+List of advanced parameters
+============================
+
+When creating or working with an Aiven for Apache Kafka service, several configuration options are available. They are summarized below:
```
Comment (ftisiot) [inline docs/products/kafka/reference/advanced-params.rst:4]: Should we add a mention on how to use these settings? By adding the `-c` configuration flag during Service creation (https://developer.aiven.io/docs/tools/cli/service.html#avn-service-create) or update (https://developer.aiven.io/docs/tools/cli/service.html#avn-service-update)

**Review by ftisiot — CHANGES_REQUESTED** [review]
Small note about taking these parameters into use, by adding links to the avn cli relevant sections

File: docs/products/kafka/reference/advanced-params.rst (hunk 1 above)
Comment (lornajane) [reply docs/products/kafka/reference/advanced-params.rst:4]: My feeling was that the list was too long, it's also in ALL the tools, the API, terraform, etc. However if we did ever build something like a CLI cheat sheet for a product, we could signpost that from this page. This is reference, so likely that other pages would link to here, if that makes sense.

**Comment by lornajane** [comment]
An excellent point from @dbeech that these "objects" are in fact very rich and structured data payloads in their own right. We should try to drill into the available API response to get more information on the fields inside the objects.

**Comment by lornajane** [comment]
I think this still needs some formatting work (I dislike the bullet points) but I switched from table to sort of definition list style to avoid trying to fit the nested properties into a table cell. Any thoughts, anyone?

**Comment by anelook** [comment]
I also feel that the bullet points are redundant. 

Also, for returned values, would it make sense to add ":" or → to indicate a returned value?

**Comment by lornajane** [comment]
@anelook can you suggest an alternative to the bullet points? I struggled to get a newline that still groups the description with the field itself. I'm also not sure what you mean by a returned value so I think I need an example if possible!

**Comment by anelook** [comment]
Added a new commit where I removed the bullet points completely without replacing with something else. Structure still looks same, just no wasted space.
I added also a symbol " =>" to indicate the returned type.

@lornajane Do you think this looks better? Otherwise we can revert last commit.

**Comment by lornajane** [comment]
I think this works pretty well! The data types aren't returned values, they're a description of what sort of data the field is. Does anyone want to approve and merge? I've just rebased to sort out the conflict in the nav file (we added too much great stuff at once, best problem ever)

**Review by anelook — APPROVED** [review]

**Comment by anelook** [comment]
Approved and merging. But if you feel "=>" is not giving any value - I can remove it with a new PR :)

**Comment by anelook** [comment]
Oh, wait, can't merge, @ftisiot  has to approve it!

**Review by ftisiot — APPROVED** [review]
Looks Good!

**Pull request merged**